test/StreamableBodyTest.cpp

```diff
@@ -72,13 +72,13 @@ TEST(StreamableBodyTest, TestGeneratorReturnsBufferedResults) {
 
   auto result = writer.get(errorCode);
   ASSERT_EQ(errorCode, boost::system::error_code());
-  ASSERT_NE(result, boost::none);
+  ASSERT_TRUE(result != boost::none);
   ASSERT_EQ(toStringView(result->first), std::string(BUFFER_SIZE, 'A'));
   ASSERT_TRUE(result->second);
 
   auto result2 = writer.get(errorCode);
   ASSERT_EQ(errorCode, boost::system::error_code());
-  ASSERT_NE(result, boost::none);
+  ASSERT_TRUE(result2 != boost::none);

```

Comment:
For some unknown reason the ASSERT_NE/EQ makros cannot be used to compare boost::none with boost::optional. It seems like gtest has a problem trying to instantiate a templated function that produces a string representation of those types

Reply:
I'll try this out, if this is a bug.